homeassistant/components/iammeter/config_flow.py

```diff
@@ -46,18 +45,20 @@ def _host_in_configuration_exists(self, host) -> bool:
         return False
 
     async def _test_connection(self, host, port):
+        errors = {}
         try:
             with async_timeout.timeout(PLATFORM_TIMEOUT):
                 await iammeter.real_time_api(host, port)
                 return True
-        except (IamMeterError, asyncio.TimeoutError):
+        except (IamMeterError, asyncio.TimeoutError) as err:
             _LOGGER.error("Device is not ready!")
-            self._errors[CONF_NAME] = "cannot_connect"
+            errors[CONF_NAME] = "cannot_connect"
+            raise PlatformNotReady from err
```

Comment:
Please raise a new specific error here. Please run our scaffold script to generate best practice code including config flow tests.